src/components/HardwareWalletConnect.js

```diff
@@ -0,0 +1,139 @@
+import React, { Fragment } from 'react';
+
+import lang from 'languages';
+import styled from 'styled-components';
+import { Button, Flex } from '@makerdao/ui-components-core';
+import { ReactComponent as TrezorLogo } from 'images/trezor.svg';
+import { ReactComponent as LedgerLogo } from 'images/ledger.svg';
+
+import useMaker from 'hooks/useMaker';
+import useModal from 'hooks/useModal';
+
+import { AccountTypes } from '../utils/constants';
+import { navigation } from '../index';
+import { mixpanelIdentify } from 'utils/analytics';
+
+const StyledLedgerLogo = styled(LedgerLogo)`
+  margin-top: -5px;
+  margin-bottom: -5px;
+`;
+
+const StyledTrezorLogo = styled(TrezorLogo)`
+  margin-top: -5px;
+  margin-bottom: -5px;
+`;
+
+const TREZOR_PATH = "44'/60'/0'/0/0";
+const DEFAULT_ACCOUNTS_PER_PAGE = 25;
+
+const onConfirm = async (maker, closeModal, accType) => {
+  const connectedAddress = maker.currentAddress();
+
+  mixpanelIdentify(connectedAddress, accType);
+
+  const { network } = navigation.receivedRoute.url.query;
+  const addressToView = connectedAddress;
+  navigation.history.push({
+    pathname: `owner/${addressToView}`,
+    search: `?network=${network}`
+  });
+
+  closeModal();
+};
+
+const renderAccountsSelection = ({
```

Comment:
@sirromdev note that the logic for actually adding a MetaMask account to the sdk [now lives in the useMaker hook](https://github.com/makerdao/mcd-cdp-portal/blob/4f53b2cc573032de31d66aeb473fff6f6e6a98c5/src/hooks/useMaker.js#L35). Do you see a clean way to do something similar for the hw wallets?

Reply:
Yes, that would clean things up. I'll have a think about it!